test/extensione2e/kafka/kafka_channel_ksvc_test.go

```diff
@@ -82,12 +84,21 @@ func TestSourceToKafkaChanelToKnativeService(t *testing.T) {
 	client := test.SetupClusterAdmin(t)
 	cleanup := func() {
 		test.CleanupAll(t, client)
-		client.Clients.KafkaChannel.MessagingV1beta1().KafkaChannels(testNamespace).Delete(context.Background(), kafkaChannelName, metav1.DeleteOptions{})
-		client.Clients.Eventing.MessagingV1().Subscriptions(testNamespace).Delete(context.Background(), subscriptionName, metav1.DeleteOptions{})
 		client.Clients.Eventing.SourcesV1beta1().PingSources(testNamespace).Delete(context.Background(), pingSourceName, metav1.DeleteOptions{})
+		if err := waitForPingSourceDeleted(client, pingSourceName); err != nil {
+			t.Errorf("PingSource not deleted in time: %v", err)
+		}
+		client.Clients.Eventing.MessagingV1().Subscriptions(testNamespace).Delete(context.Background(), subscriptionName, metav1.DeleteOptions{})
+		if err := waitForSubscriptionDeleted(client, subscriptionName); err != nil {
+			t.Errorf("Subscription not deleted in time: %v", err)
+		}
+		client.Clients.KafkaChannel.MessagingV1beta1().KafkaChannels(testNamespace).Delete(context.Background(), kafkaChannelName, metav1.DeleteOptions{})
+		if err := waitForChannelDeleted(client, kafkaChannelName); err != nil {
+			t.Errorf("Channel not deleted in time: %v", err)
+		}
+
```

Comment:
I think a better and more commonly used solution (for the sake of our e2e tests) might be to add a random suffix to the resources we create so there isn't a collision between runs, instead of failing due to the infra not keeping up with the tests harness and creating more noise.  wdyt?

Reply:
You're probably right. This change would need a little more time. Maybe we could merge this and next week I could implement the better solution? Let's see if this change passes CI.

Reply:
Pushed one more commit which removes the wait time because it didn't help anyway. So, now the PR just removes unnecessary cleanups.